src/console/Microsoft.Health.Fhir.Ingest.Console.FhirTransformation/Startup.cs

```diff
@@ -0,0 +1,136 @@
+using Microsoft.Health.Extensions.Fhir.Service;
+using Microsoft.Health.Fhir.Client;
+using Microsoft.Health.Fhir.Ingest.Config;
+using Microsoft.Health.Fhir.Ingest.Console.Common;
+using Microsoft.Health.Fhir.Ingest.Host;
+using Microsoft.Health.Fhir.Ingest.Service;
+using Microsoft.Health.Fhir.Ingest.Telemetry;
+using Microsoft.Health.Fhir.Ingest.Template;
+using Microsoft.Health.Logging.Telemetry;
+
+namespace Microsoft.Health.Fhir.Ingest.Console.FhirTransformation
+{
+    public class Startup : StartupBase
+    {
+        public Startup()
+            : base()
+        {
+        }
+
+        public override string ApplicationType => Common.ApplicationType.MeasurementToFhir;
+
+        public override string OperationType => ConnectorOperation.FHIRConversion;
+
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            base.ConfigureServices(services);
+
+            Configuration.GetSection("FhirService")
+                .GetChildren()
+                .ToList()
+                .ForEach(env => Environment.SetEnvironmentVariable(env.Path, env.Value));
+
+            // for open source, use default azure credential
+            services.AddSingleton<IAzureCredentialProvider, AzureCredentialProvider>();
+
+            services.Configure<ResourceIdentityOptions>(Configuration.GetSection("ResourceIdentity"));
+            services.Configure<ObservationCacheOptions>(Configuration.GetSection(ObservationCacheOptions.Settings));
+            services.AddSingleton<IFhirTemplateProcessor<ILookupTemplate<IFhirTemplate>, Observation>, R4FhirLookupTemplateProcessor>();
+            services.AddSingleton<IFactory<IFhirClient>, FhirClientFactory>();
+            services.AddSingleton<IFhirService, FhirService>();
+            services.AddSingleton<FhirImportService, R4FhirImportService>();
+            services.AddSingleton<IExceptionTelemetryProcessor, FhirExceptionTelemetryProcessor>();
+            services.AddSingleton<MeasurementFhirImportOptions>();
+            MeasurementImportServiceExtensions.AddImportService(services, Configuration);
+            services.AddSingleton(sp => sp.GetRequiredService<IFactory<IFhirClient>>().Create());
+
+            services.AddSingleton(sp => Options.Create(new FhirClientFactoryOptions { UseManagedIdentity = true }));
+
+            services.AddSingleton(sp => {
+                // Get IAzureExternalIdentityCredentialProvider if it exists, else use IAzureCredentialProvider
+                var externalMiTokenProvider = sp.GetService<IAzureExternalIdentityCredentialProvider>();
+                var serviceMiTokenProvider = sp.GetService<IAzureCredentialProvider>();
+                var tokenProvider = externalMiTokenProvider ?? serviceMiTokenProvider;
+                return Options.Create(new FhirClientFactoryOptions { CredentialProvider = tokenProvider });
+            });
+
+            // services.AddSingleton<IHealthChecker, FhirServiceHealthChecker>();
```

Comment:
We should remove commented code before checking in.

Reply:
updated